crates/dids/src/method/jwk.rs

```diff
@@ -0,0 +1,120 @@
+use crate::did::Did;
+use crate::method::{DidMethod, DidMethodError};
+use crate::resolver::DidResolutionResult;
+use async_trait::async_trait;
+use crypto::key::{Key, KeyType};
+use crypto::key_manager::KeyManager;
+use did_jwk::DIDJWK as SpruceDidJwkMethod;
+use ssi_dids::did_resolve::{DIDResolver, ResolutionInputMetadata};
+use ssi_dids::{DIDMethod, Source};
+use std::sync::Arc;
+
+/// Concrete implementation for a did:jwk DID
+pub struct DidJwk {
+    uri: String,
+    key_manager: Arc<dyn KeyManager>,
+}
+
+impl Did for DidJwk {
+    fn uri(&self) -> &str {
+        &self.uri
+    }
+
+    fn key_manager(&self) -> &Arc<dyn KeyManager> {
+        &self.key_manager
+    }
+}
+
+/// Options that can be used to create a did:jwk DID
+pub struct DidJwkCreateOptions {
+    pub key_type: KeyType,
```

Comment:
I actually found `KeyType` to be a bit confusing. Looking at the docs for `KeyType` didn't actually help. 

Is `KeyType` meant to represent https://www.rfc-editor.org/rfc/rfc7518.html#section-6.1 ? Or is it meant to represent the name of the [elliptic curve](https://www.iana.org/assignments/jose/jose.xhtml#web-key-elliptic-curve)?

In kotlin, we defined the creation options as Algo and Curve, curious how/why this diverged?

Reply:
I originally had it as `KeyAlgorithm`, but was recommended to rename it to `KeyType` here: https://github.com/TBD54566975/web5-rs/pull/17#discussion_r1401124839

Each of the algorithms use the same curve in Kotlin, so I just didn't add them to the Rust version yet. Can add them later if/when we use them.

Reply:
yes I was referring to the concept similar to `kty` https://www.rfc-editor.org/rfc/rfc7518.html#section-6.1